common/djangoapps/student/models.py

```diff
@@ -1707,8 +1708,9 @@ def upgrade_deadline(self):
             )
             return None
 
+        schedule_config = CourseScheduleConfiguration.current(self.course_id)
         try:
-            if self.schedule:
+            if schedule_config.enabled and schedule_config.verified_upgrade_deadline_enabled and self.schedule:
```

Comment:
What's the need for this? If there is a `Schedule`, why would we not use it? The creation of a schedule is already dependent on this configuration.